src/modules/contracts/entities/contracts.entity.ts

```diff
@@ -0,0 +1,43 @@
+import { Freelancer } from 'src/modules/freelancer/entities/freelancer.entity';
+import { JobPost } from 'src/modules/job-post/entities/job-post.entity';
+import {
+  Column,
+  Entity,
+  ManyToOne,
+  OneToOne,
+  PrimaryGeneratedColumn,
+} from 'typeorm';
+
+export enum ContractStatus {
+  PENDING = 'pending',
+  ACTIVE = 'active',
+  CLOSED = 'closed',
+}
+
+@Entity()
+export class Contracts {
+  @PrimaryGeneratedColumn()
+  id: number;
+
+  @OneToOne(() => JobPost, (jobPost) => jobPost.contract)
+  jobPost: JobPost;
+
+  @ManyToOne(() => Freelancer, (freelancer) => freelancer.id, {
+    cascade: true,
+    nullable: true,
+  })
+  freelancer: Freelancer;
+
+  @Column({
+    type: 'enum',
+    enum: ContractStatus,
+    default: ContractStatus.PENDING,
+  })
+  contractStatus: ContractStatus;
```

Comment:
I dont think that we need status, if end date exist contract close, if end date not exist contract opened

Reply:
You're right, but what do you think about the logic below? Or would you prefer only open and closed states to be shown in the front end?

If there's no starting date, the contract will be in the 'pending' or 'analyzing bids' state. When a proposal and job offer are accepted, a starting date will be added and the contract will be active. When the job is done, an ending date will be added and the contract will be closed

Reply:
Removed the contract status 👍🏻

Reply:
I think you confused between offer and contract, for me its 2 totally different entities, as contract become from accepted offer. to I'm not fully understand how you want to work with this logic

Reply:
What we want right now is to display to the client in his job post information about the contract, if it is pending (no job offers accepted), open (one job offer was accepted), or if it was closed. The freelancer will also be able to see on his 'My Contracts' page all contracts that are associated with him. The job offer entity will be done later.

So in the future, once the job offer is accepted by the freelancer, the contract start date will be updated, and when everything is done or the client closes his job post, the end date will be updated.

Reply:
there no pending status, contract or open or close